### PR TITLE
docs: Reorganize tutorials and guides.

### PR DESCRIPTION
**Pull request by max-radin**
## Description

This PR makes some improvements to the organization and formatting of the docs.
* Removed "Integrations, modules, & libraries" section since the content is now covered by the new guides.
* Contributing and Package Structure sections moved from tutorials to guides.
* Converted bulleted lists to tables.
* Misc formatting tweaks.

## Please verify that you have completed the following steps

- [x] I have self-reviewed my code.
- [x] I have included test cases validating introduced feature/fix.
- [x] I have updated documentation.

### CODE REVIEW

File: docs/guides/interfaces.rst [hunk 1]
```diff
@@ -38,28 +34,32 @@ However, using abstract classes requires using inheritance, which sometimes is u
...
+You might wonder - what if I have two optimizers which take different sets of parameters? For example, you might want to use `L-BFGS-B <https://docs.scipy.org/doc/scipy/reference/optimize.minimize-lbfgsb.html#optimize-minimize-lbfgsb>`_ and `COBYLA <https://docs.scipy.org/doc/scipy/reference/optimize.minimize-cobyla.html#optimize-minimize-cobyla>`_ from ``scipy``. This is how you would use them in scipy: 
 
 .. literalinclude:: ../examples/guides/interfaces_guide.py
     :start-after: >> Guide code snippet: script showing scipy optimizers
     :lines: 2-3
     :language: python
 
 
-`L-BFGS-B` doesn't support argument called `constraints` and `COBYLA` doesn't support `bounds`. So how can you make them fit the same interface, if they have different inputs?
+``L-BFGS-B``` doesn't support argument called ``constraints`` and ``COBYLA`` doesn't support ``bounds``. So how can you make them fit the same interface, if they have different inputs?
```
Comment (mstechly) [inline docs/guides/interfaces.rst:60]: I think this is one too many "`"?

File: docs/guides/interfaces.rst [hunk 2]
```diff
@@ -38,28 +34,32 @@ However, using abstract classes requires using inheritance, which sometimes is u
...
 
-In the case of optimizers, Orquestra solves this problem by moving differing parameters into object initialization. Constructed optimizers provide :meth:`minimize <orquestra.opt.api.Optimizer.minimize>` method that always accepts only :func:`create_cost_function <orquestra.vqa.cost_function.cost_function.create_cost_function>` and `initial_params` arguments. Once constructed, optimizers are fully interchangeable.
+In the case of optimizers, Orquestra solves this problem by moving differing parameters into object initialization. Constructed optimizers provide :meth:`minimize <orquestra.opt.api.Optimizer.minimize>` method that always accepts only :func:`create_cost_function <orquestra.vqa.cost_function.cost_function.create_cost_function>` and ``initial_param`s`` arguments. Once constructed, optimizers are fully interchangeable.
```
Comment (mstechly) [inline docs/guides/interfaces.rst:62]: I think something is wrong here – the name of the arg is `initial_params`, not `initial_param`?

**Review by mstechly — APPROVED** [review]
Two minor comments, but approving.
Also, after @dexter2206 merged our changes to VQA guides, there are some conflicts with `main`.

File: docs/guides/interfaces.rst (hunk 1 above)
Comment (max-radin) [reply docs/guides/interfaces.rst]: Good catch, fixed in faec13157e2adf175e2306705f526a4ec977cca2

File: docs/guides/interfaces.rst (hunk 2 above)
Comment (max-radin) [reply docs/guides/interfaces.rst]: An extra backtick snuck in here, now fixed in faec13157e2adf175e2306705f526a4ec977cca2

**Pull request merged**